frontend/packages/console-app/src/components/modals/restore-pvc/restore-pvc-modal.tsx

```diff
@@ -85,19 +84,17 @@ const RestorePVCModal = withHandlePromise<RestorePVCModalProps>(
       PersistentVolumeClaimKind
     >(PersistentVolumeClaimModel, resource?.spec?.source?.persistentVolumeClaimName, namespace);
 
-    const [
-      snapshotClassResource,
-      snapshotClassResourceLoaded,
-      snapshotClassResourceLoadError,
-    ] = useK8sGet<VolumeSnapshotClassKind>(
-      VolumeSnapshotClassModel,
-      resource?.spec?.volumeSnapshotClassName,
+    const pvcStorageClassName = pvcResource?.spec?.storageClassName;
+    const [scResource, scResourceLoaded, scResourceLoadError] = useK8sGet<StorageClassResourceKind>(
+      StorageClassModel,
+      pvcStorageClassName,
     );
 
     const [volumeMode, setVolumeMode] = React.useState('');
     const onlyPvcSCs = (scObj: StorageClassResourceKind) =>
```

Comment:
this onlyPvcSCs logic is used in clone as well as restore, why don't we move this function as common util function?

Reply:
why selectors?